src/components/content/services/contentSrv.js

```diff
@@ -127,6 +127,18 @@
                 });
             };
 
+            ContentSrv.getAllIdsByKey = function(key) {
+                var idsByKeys;
+                var arrayOfKeys;
+                return contentDataFunc().get().then(function(dataObj) {
+                    idsByKeys = Object.keys(dataObj.latestRevisions);
```

Comment:
It will be much faster to iterate over latestRevisions object rather than retrieving its keys and then iterating over them.